server/helpers/wowCommunityApi.js

```diff
@@ -36,35 +37,163 @@ const get = (url, metricLabels) => {
     onBeforeAttempt: () => {
       end = blizzardApiResponseLatencyHistogram.startTimer(metricLabels);
     },
+    onFailure: (err, attempt) => {
+      if (err.statusCode === 401 && attempt === 1) {
+        // if the request in unauthorized, try renewing
+        // the battle net creds, only worth trying once though
+        return getAccessToken(region);
```

Comment:
This doesn't try renewing if it already has an access token, I think you need to reset it first.

I don't think there's any need to only do it for the first attempt, in fact if the first attempt fails due to a connection error or something like that, subsequent requests might run into this.

Reply:
you're right and I noticed on some testing, I'm allowing it to pass back changes options, you can tell me if you see anything that can go wrong there?